SecretManagement.LastPass.Extension/SecretManagement.LastPass.Extension.psm1

```diff
@@ -174,51 +202,51 @@ function Set-Secret
-                $Secret.$k = [System.Runtime.InteropServices.Marshal]::PtrToStringAuto($BSTR)
+                $Secret.$k = [System.Net.NetworkCredential]::new("", $Secret.$k).Password
             }
             $sb.AppendLine("$($k): $($Secret.$k)") | Out-Null
         }
 
         # Notes need to be on a new line
         if ($null -ne $Secret.Notes) {
             if ($Secret.Notes -is [securestring]) {
-                $BSTR = [System.Runtime.InteropServices.Marshal]::SecureStringToBSTR($Secret.Notes)
-                $Secret.Notes = [System.Runtime.InteropServices.Marshal]::PtrToStringAuto($BSTR)
+                $Secret.Notes = [System.Net.NetworkCredential]::new("", $Secret.Notes).Password
             }
             $sb.AppendLine("Notes: `n$($Secret.Notes)") | Out-Null
         }
     } 
     
-    try {
-        $res = Invoke-lpass 'show', '--sync=now', '--name', $Name, '2>/dev/null' -ErrorAction Stop 
-        $SecretExists = $null -ne $res 
-
-        if ($SecretExists) {
-            Write-Verbose "Editing secret" 
-            $sb.ToString() | Invoke-lpass 'edit', '--non-interactive', $Name
-        } else {
-            Write-Verbose "Adding new secret" 
-            $NoteTypeArgs = @()
-            if ($null -ne $Secret.NoteType) {
-                if ($Secret.NoteType -is [securestring]) { 
-                    $BSTR = [System.Runtime.InteropServices.Marshal]::SecureStringToBSTR($Secret.NoteType)
-                    $Secret.NoteType = [System.Runtime.InteropServices.Marshal]::PtrToStringAuto($BSTR)
-                }
-                $NoteTypeArgs += "--note-type=$($Secret.NoteType)"
+    $res = Invoke-lpass 'show', '--sync=now', '--name', $Name
+
+    # We use ToString() here to turn the ErrorRecord into a string if we got an ErrorRecord
+    $SecretExists = if ($null -eq $res) {
+        # This should never ever happen...
+        Write-Warning "Querying the secret $Name produced an unexpected result of `$Null"
+        $false
+    } elseif ($res.ToString() -eq $lpassMessage.AccountNotFound) {
+        $false
+    } else {
+        $true
+    }
+
+    if ($SecretExists) {
+        Write-Verbose "Editing secret"
+        $sb.ToString() | Invoke-lpass 'edit', '--non-interactive', $Name
+    } else {
+        Write-Verbose "Adding new secret"
+        $NoteTypeArgs = @()
+        if ($null -ne $Secret.NoteType) {
+            if ($Secret.NoteType -is [securestring]) { 
+                $Secret.NoteType = [System.Net.NetworkCredential]::new("", $Secret.NoteType).Password
             }
-            $sb.ToString() | Invoke-lpass 'add', $Name, '--non-interactive', $NoteTypeArgs
-            #Explicit sync so calling set again do not duplicate the secret (add --sync=now not fast enough)
-            Invoke-lpass 'sync' 
+            $NoteTypeArgs += "--note-type=$($Secret.NoteType)"
         }
-       
+        $sb.ToString() | Invoke-lpass 'add', $Name, '--non-interactive', $NoteTypeArgs
```

Comment:
```suggestion
        $res = $sb.ToString() | Invoke-lpass 'add', $Name, '--non-interactive', $NoteTypeArgs
        if ($null -ne $res)  {
            Write-Error ($res | Out-String)
            return $false
        }
```

Not quite sure that would be the best implementation though but since we suppress the error by default, don't you want to expose the error if it occurs. 

Take for instance this : 

![image](https://user-images.githubusercontent.com/1980296/99024261-30ab6c80-2534-11eb-9614-9cc82d3f6203.png)

If we do nothing, add secret will succeed... 
Even though we received an error from the call. 

To reproduce. 
- Create a secret with a custom type in lastpass (eg: credit card) 
Then 

```
$Boop = Get-Secret -Name "My Credit Card" -AsPlainText -vault lpt -Verbose
$Boop.NoteType = 'sdsd'
Set-Secret -Name 'My Credit Card dupe '  -Secret $Boop -Vault lpt -Verbose
```